### DISCUSSION TITLE
Clockplot: visualizing sleep data around the clock!

### DISCUSSION THREAD

**Discussion by marcodallavecchia** [has2k1/plotnine#812] [2024 Plotnine Contest]
### Authors

Marco Dalla Vecchia, Valeria Fonseca Diaz

### Links

[mvgroup](https://gitlab.com/mvgroup), [mvlearn](https://mvlearn.netlify.app/)

### Full description

# Clockplot: visualizing sleep data around the clock!
![image](https://github.com/user-attachments/assets/bd736f98-0a5d-4ffa-b4f0-1f51e97d9727)

Read the article [here](https://mvgroup.gitlab.io/visionlization/data_visualization/2024-plotnine-contest/)!

## Motivation
We were inspired by the classic concept of the alarm clock, and its association to the dreaded waking up time. We exported some personal sleep data from a Mi Band sportswear which records an estimated start and end time of sleep (as well as some other data not used in this project). We thought it would be interesting to visualize our sleep habits around the clock. We thought the most interesting information to be the sleep duration, start and end together with how often in the year we would go to sleep at specific times.

## Explanation
The clockplot shows arcs starting and ending at times corresponding to falling asleep and waking up times. The distance of the arc from the center of the clock represents the ranking of that sleep interval and each arc is colored by the frequency of that habit. The hands of the clock depict the approximated mean waking up time.

### Code repository

Check out the code, documentation and instructions [here](https://gitlab.com/mvgroup/visionlization/data_visualization/2024-plotnine-contest)

**Comment by has2k1** [comment]
![clockplot](https://github.com/user-attachments/assets/792ad3a7-1374-430b-a84b-a3986c00f17e)

**Comment by EconomiCurtis** [comment]
Ahoy @marcodallavecchia , 
I am one of the contest organizers. 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Reply by marcodallavecchia** [reply]
Hey! Thanks a lot for the gift, participating was awesome! I filled in the form :)

**Comment by Gchism94** [comment]
It would be great to see polar coordinate functionality built into plotnine, perhaps using your work as a starting point. Nice work!

**Reply by marcodallavecchia** [reply]
Thanks for the message! Indeed, we were really hoping to build it up using built-in polar coordinates in the first place. Unfortunately, it seems to be a fairly seasoned and craved [issue](https://github.com/has2k1/plotnine/issues/10) with the current plotnine library is still missing.

**Comment by has2k1** [comment]
Marco Dalla Vecchia and Valeria Fonseca Diaz, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.
